lib/ClangImporter/ImportDecl.cpp

```diff
@@ -9236,8 +9248,10 @@ Decl *ClangImporter::Implementation::importDeclAndCacheImpl(
   startedImportingEntity();
   Decl *Result = importDeclImpl(ClangDecl, version, TypedefIsSuperfluous,
                                 HadForwardDeclaration);
-  if (!Result)
+  if (!Result) {
+    ImportedDecls[{Canon, version}] = None;
     return nullptr;
+  }
```

Comment:
This is the cache the failed decls. Is there anything else that should be cached here or anywhere else other than the Canon and version?